### PR TITLE
[MWPW-161870] Fixing the issue that sidenav can't be collapsed.

### PR DESCRIPTION
**Pull request by seanchoi-dev**
Resolves:
- [MWPW-161870](https://jira.corp.adobe.com/browse/MWPW-161870)
- [MWPW-162024](https://jira.corp.adobe.com/browse/MWPW-162024)

**Test URLs:**
- Before: https://main--milo--adobecom.aem.page/?martech=off
- After: https://MWPW-161870--milo--adobecom.aem.page/?martech=off

**Test for consumer site:**
- Before: https://main--cc--adobecom.hlx.page/products/catalog
- After: https://main--cc--adobecom.hlx.page/products/catalog?milolibs=MWPW-161870#category=creativity-design

### CODE REVIEW

File: libs/features/mas/web-components/src/sidenav/merch-sidenav-list.js [hunk 1]
```diff
@@ -91,6 +88,9 @@ export class MerchSidenavList extends LitElement {
             );
             if (!element) return;
             this.updateComplete.then(() => {
+                if (element.firstElementChild?.tagName === 'SP-SIDENAV-ITEM') {
```
Comment (Axelcureno) [inline libs/features/mas/web-components/src/sidenav/merch-sidenav-list.js:91]: Nit: Move string to a `const`

File: libs/features/mas/web-components/test/sidenav/catalog-sidenav.test.html.js [hunk 2]
```diff
@@ -196,12 +196,14 @@ runTests(async () => {
 
         it('does refresh from url state', async () => {
             const search = document.querySelector('sp-search');
+            const sidenavItemCD = document.querySelector('sp-sidenav-item[label="Creativity And Design"]');
             search.value = '';
-            pushState({ search: 'photoshop' });
+            pushState({ search: 'photoshop', filter: 'creativitydesign' });
```
Comment (Axelcureno) [inline libs/features/mas/web-components/test/sidenav/catalog-sidenav.test.html.js:201]: Nit: Move string to a `const`.

**Review by Axelcureno — APPROVED** [review]

**Comment by Roycethan** [comment]
@seanchoi-dev  I checked your fix link provided here: https://main--cc--adobecom.hlx.live/products/catalog?milolibs=MWPW-161870
1)Content on a particular filter should expands and collapse with keyboard---This works now
2) Expanding content user should hear updated results since the page content changed – User(Screen Reader) is still not hearing updated results upon expanding a filter link.

**Comment by seanchoi-dev** [comment]
> @seanchoi-dev I checked your fix link provided here: https://main--cc--adobecom.hlx.live/products/catalog?milolibs=MWPW-161870 1)Content on a particular filter should expands and collapse with keyboard---This works now 2) Expanding content user should hear updated results since the page content changed – User(Screen Reader) is still not hearing updated results upon expanding a filter link.

I committed new code and they should work for both tickets:
- [MWPW-161870](https://jira.corp.adobe.com/browse/MWPW-161870)
- [MWPW-162024](https://jira.corp.adobe.com/browse/MWPW-162024)

**Comment by Roycethan** [comment]
> > @seanchoi-dev I checked your fix link provided here: https://main--cc--adobecom.hlx.live/products/catalog?milolibs=MWPW-161870 1)Content on a particular filter should expands and collapse with keyboard---This works now 2) Expanding content user should hear updated results since the page content changed – User(Screen Reader) is still not hearing updated results upon expanding a filter link.
> 
> I committed new code and they should work for both tickets:
> 
> * [MWPW-161870](https://jira.corp.adobe.com/browse/MWPW-161870)
> * [MWPW-162024](https://jira.corp.adobe.com/browse/MWPW-162024)

@seanchoi-dev 
Few issues:
1) Search with text "ace" gives " Your search for ace did not yield any results" -Screen reader doesn't read this 
2) Clicking on filter say "Creativity and Design"   gives "34 results in Creativity and Design" - Screen reader reads "34 results in Creativity and Design" twice.

**Comment by seanchoi-dev** [comment]
> > > @seanchoi-dev I checked your fix link provided here: https://main--cc--adobecom.hlx.live/products/catalog?milolibs=MWPW-161870 1)Content on a particular filter should expands and collapse with keyboard---This works now 2) Expanding content user should hear updated results since the page content changed – User(Screen Reader) is still not hearing updated results upon expanding a filter link.
> > 
> > 
> > I committed new code and they should work for both tickets:
> > 
> > * [MWPW-161870](https://jira.corp.adobe.com/browse/MWPW-161870)
> > * [MWPW-162024](https://jira.corp.adobe.com/browse/MWPW-162024)
> 
> @seanchoi-dev Few issues:
> 
> 1. Search with text "ace" gives " Your search for ace did not yield any results" -Screen reader doesn't read this
> 2. Clicking on filter say "Creativity and Design"   gives "34 results in Creativity and Design" - Screen reader reads "34 results in Creativity and Design" twice.

As I tested with "ace" and Clicking on filter "Creativity and Design", it works as expected to me.
Can you test it again?

**Comment by Roycethan** [comment]
> https://main--cc--adobecom.hlx.live/products/catalog?milolibs=MWPW-161870

@seanchoi-dev 
Attaching screen video here: (plz enable audio while running)

Uploading screenRecording-2-12-2024-16-1.mp4…


1) I think i see whats gping on here for " Your search for ace did not yield any results".. so typing "ace" and either stay without doing any action or cliking "Enter" on keyboard doesn't read by Screen reader, only cliking "Space bar" or clikcing on the result element - is read by screen reader.
2) This is also reproducible - the result text is read twice

**Comment by seanchoi-dev** [comment]
> > https://main--cc--adobecom.hlx.live/products/catalog?milolibs=MWPW-161870
> 
> @seanchoi-dev Attaching screen video here: (plz enable audio while running)
> 
>  screenRecording-2-12-2024-16-1.mp4 
> 1. I think i see whats going on here for " Your search for ace did not yield any results".. so typing "ace" and either stay on screen without doing any action or click"Enter" on keyboard which doesn't read by Screen reader (which usually performed typically by users)
>    only clicking "Space bar" or clicking on the resultant element - is read by screen reader.
> 2. This is also reproducible - the result text is read twice

I can't enable the audio from your recording but I guess you are showing the issue you're mentioning.
I using voiceover for testing and it works fine to me. What tool do you use for screen reader?

**Comment by Roycethan** [comment]
> * MWPW-161870



> > > https://main--cc--adobecom.hlx.live/products/catalog?milolibs=MWPW-161870
> > 
> > 
> > @seanchoi-dev Attaching screen video here: (plz enable audio while running)
> > screenRecording-2-12-2024-16-1.mp4
> > 
> > 1. I think i see whats going on here for " Your search for ace did not yield any results".. so typing "ace" and either stay on screen without doing any action or click"Enter" on keyboard which doesn't read by Screen reader (which usually performed typically by users)
> >    only clicking "Space bar" or clicking on the resultant element - is read by screen reader.
> > 2. This is also reproducible - the result text is read twice
> 
> I can't enable the audio from your recording but I guess you are showing the issue you're mentioning. I'm using Mac's voiceover for testing and it works fine to me. What tool do you use for screen reader?

@seanchoi-dev  
I'm using NVDA on Windows 11 (note: available on browser-stack) since the audit is performed using this , also me mentioned in tickets...

**Comment by seanchoi-dev** [comment]
> > * MWPW-161870
> 
> > > > https://main--cc--adobecom.hlx.live/products/catalog?milolibs=MWPW-161870
> > > 
> > > 
> > > @seanchoi-dev Attaching screen video here: (plz enable audio while running)
> > > screenRecording-2-12-2024-16-1.mp4
> > > 
> > > 1. I think i see whats going on here for " Your search for ace did not yield any results".. so typing "ace" and either stay on screen without doing any action or click"Enter" on keyboard which doesn't read by Screen reader (which usually performed typically by users)
> > >    only clicking "Space bar" or clicking on the resultant element - is read by screen reader.
> > > 2. This is also reproducible - the result text is read twice
> > 
> > 
> > I can't enable the audio from your recording but I guess you are showing the issue you're mentioning. I'm using Mac's voiceover for testing and it works fine to me. What tool do you use for screen reader?
> 
> @seanchoi-dev I'm using NVDA on Windows 11 (note: available on browser-stack) since the audit is performed using this , also mentioned in tickets...

This is very likely NVDA's bug. There are bunch of double-speaking bug and other bugs reported.
(https://github.com/nvaccess/nvda/labels/bug%2Fdouble-speaking)
I am currently trying to find a different way to perform same thing without the bug now.

Interestingly, the bug shows only under specific condition. I tried with different word and different steps and sometimes reproducible sometimes not.

**Comment by seanchoi-dev** [comment]
@Roycethan ,
I've fixed the doubling speaking issue on NVDA app, but haven't found a good solution the not reading for no search result w/o space. I confirmed that the bug exists only for NVDA, though. Do you feel it is crucial for preventing merging this?

**Comment by Roycethan** [comment]
> @Roycethan , I've fixed the doubling speaking issue on NVDA app, but haven't found a good solution the not reading for no search result w/o space. I confirmed that the bug exists only for NVDA, though. Do you feel it is crucial for preventing merging this?

@seanchoi-dev signed off for https://jira.corp.adobe.com/browse/MWPW-161870 and mentioned above issue in https://jira.corp.adobe.com/browse/MWPW-162024 asking Creator.. 
Meanwhile moving this to Stage.

**Pull request merged**